lib/membrane_h264_plugin/parser.ex

```diff
@@ -181,16 +181,20 @@ defmodule Membrane.H264.Parser do
         stream_structure -> stream_structure
       end
 
-    au_timestamp_generator =
-      if opts.generate_best_effort_timestamps,
-        do: AUTimestampGenerator.new(opts.generate_best_effort_timestamps),
-        else: nil
+    {au_timestamp_generator, framerate} =
+      if opts.generate_best_effort_timestamps do
+        {AUTimestampGenerator.new(opts.generate_best_effort_timestamps),
+         opts.generate_best_effort_timestamps.framerate}
+      else
+        {nil, nil}
+      end
 
     state = %{
       nalu_splitter: nil,
       nalu_parser: nil,
       au_splitter: AUSplitter.new(),
       au_timestamp_generator: au_timestamp_generator,
+      framerate: framerate,
```

Comment:
I am just slightly worried that we have the `framerate` that might change based on the `stream_format` received, and we use it to create an output `stream_format`, and the `generate_best_effort_timestamps.framerate` that we set once, via options,  and use to generate timestamps

Reply:
The only case is when the incoming `H264` stream_format has `framerate` set to nil and `generate_best_effort_timestamps` was provided